frontend/src/buttons/DisconnectButton/DisconnectButton.tsx

```diff
@@ -13,16 +13,17 @@ type Props = {
 }
 
 export const DisconnectButton: React.FC<Props> = ({ service, size = 'medium', color = 'primary', connection }) => {
-  const hidden = !connection || !connection.active
-  const connecting = !!connection?.connecting
+  const hidden = !connection || !connection.connected
+  const disconnecting = !connection?.enabled && !!connection?.connected
   return (
     <Fade in={!hidden} timeout={600}>
       <div>
         <DynamicButton
-          title={connecting ? 'Connecting' : 'Disconnect'}
+          title={disconnecting ? 'Stopping' : 'Disconnect'}
```

Comment:
are you sure to hide the message `Connecting`?

Reply:
Yes - it appears in the other button.